src/plugins/discover/public/application/components/no_results/no_results.tsx

```diff
@@ -186,6 +187,39 @@ export const DiscoverNoResults = ({ queryString, query, savedQuery, timeFieldNam
     fetchSavedQueries();
   }, [setSavedQueries, query, savedQuery]);
 
+  useEffect(() => {
```

Comment:
There are two potential issues with the code below from the component side:

1. Compared to the old implementation, the `sampleQueries` will only be set when `query?.dataset?.type` is truthy. Is this by design? Do we need to set `sampleQueries` to `languageSampleQueries` when `query?.dataset?.type` is not defined?

2. Do we need to clear the side effect for component unmount? Since `datasetType.getSampleQueries(query.dataset, query.language)` sometimes returns a Promise object, it may be resolved after the component has unmounted. Should we add a flag to avoid calling `setSampleQueries` in this case?